software_reseller/data/project_project.xml

```diff
@@ -6,43 +6,54 @@
+        <field name="task_properties_definition" eval="[
+            {'name': 'version', 'type': 'char', 'string': 'Java Version', 'default': '20'}, 
+            {'name': 'license_key', 'type': 'char', 'string': 'License Key', 'default': 'aB9kL3pQ7xT2mH5cV8nR4jW1'}
+        ]" />
     </record>
     <record id="project_project_2" model="project.project" context="{'mail_auto_subscribe_no_notify': True}">
         <field name="name">Odoo Licenses</field>
         <field name="label_tasks">Licenses</field>
         <field name="stage_id" ref="project.project_project_stage_0" />
         <field name="allow_timesheets" eval="False" />
         <field name="x_is_license" eval="True"/>
+        <field name="x_software">Odoo</field>
         <field name="user_id" ref="base.user_admin"/>
         <field name="is_favorite" eval="False"/>
         <field name="account_id" ref="account_analytic_account_4"/>
         <field name="type_ids" eval="[(6, 0, [ref('project_task_type_1'), ref('project_task_type_2'), ref('project_task_type_3'), ref('project_task_type_4')])]" />
-        <field name="task_properties_definition" eval="[{'name': '3b2b9964284a7f63', 'type': 'many2one', 'domain': False, 'string': 'Client', 'comodel': 'res.partner', 'default': False}, {'name': '14e63001c39f0a7c', 'type': 'integer', 'string': 'Number of Users', 'default': 10}, {'name': '5238d6796e605ce9', 'type': 'selection', 'string': 'Odoo Version', 'default': False, 'selection': [['854ed77adce74cd6', 'Odoo 14'], ['5e030d963127e2d2', 'Odoo 15'], ['280c6b03bf099390', 'Odoo 16'], ['d876b1557301d0ef', 'Odoo 17']]}, {'name': '19460db267e8e4c3', 'type': 'selection', 'string': 'Hosted On', 'default': False, 'selection': [['7fc38e587064fa82', 'On Premise'], ['0acee79ccc0c1c8c', 'Odoo.sh'], ['bd1d12f1e743ece6', 'Odoo Online']]}, {'name': '33ae81dcafa5961b', 'type': 'char', 'string': 'License Key', 'default': ''}]" />
+        <field name="task_properties_definition" eval="[
+            {'name': 'number_of_users', 'type': 'integer', 'string': 'Number of Users', 'default': 10},
+            {'name': 'hosted_on', 'type': 'selection', 'string': 'Hosted On', 'default': False, 'selection': [['on_premise', 'On Premise'], ['odoo_sh', 'Odoo.sh'], ['odoo_online', 'Odoo Online']], 'default': 'odoo_online'},
+            {'name': 'license_key', 'type': 'char', 'string': 'License Key', 'default': 'BB9kL3pQ7xU2mH5cV8nR4jY1'},
+            {'name': 'version', 'type': 'selection', 'string': 'Odoo Version', 'default': False, 'selection': [['odoo_14', 'Odoo 14'], ['odoo_15', 'Odoo 15'], ['odoo_16', 'Odoo 16'], ['odoo_17', 'Odoo 17']], 'default': 'odoo_17'},
```

Comment:
is it a good idea to have a selection here? Is it easy to edit and add new ones? What about recent versions?